jsonrpc/src/common/messageReader.ts

```diff
@@ -202,13 +215,9 @@ export class ReadableStreamMessageReader extends AbstractMessageReader {
 		this.nextMessageLength = -1;
```

Comment:
This might break the whole data processing on the second listen call. If a partial message is already read that had a full header reseting this will break the message processing.

Reply:
You are correct, this needs to be moved out of the listen. Even if we go with the exception approach we may need to clean up this class regardless.